### PR TITLE
[Docs] Add code review keywords for nits, techdebt, and show stoppers

### PR DESCRIPTION
**Pull request by dylanlott**
## Description

Adds recommended keywords NIT, TECHDEBT, and SOS for use during review to signal next steps for the author.

### Summary generated by Reviewpad on 21 Jun 23 02:34 UTC
This pull request adds new code review keywords for `nits`, `techdebt`, and `show stoppers` to the `CODE_REVIEW_GUIDELINES.md` file. It also updates the formatting and provides additional guidelines for Pull Request approval.

## Issue

Stacks on top of #773

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Major breaking change
- [x] Documentation
- [ ] Other 

## List of changes

- adds definitions for NIT, TECHDEBT, TECHDEBT(XXX), and SOS keywords for use in code review.

## Testing

- [ ] `make develop_test`; if any code changes were made
- [ ] `make test_e2e` on [k8s LocalNet](https://github.com/pokt-network/pocket/blob/main/build/localnet/README.md); if any code changes were made
- [ ] `e2e-devnet-test` passes tests on [DevNet](https://pocketnetwork.notion.site/How-to-DevNet-ff1598f27efe44c09f34e2aa0051f0dd); if any code was changed
- [ ] [Docker Compose LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md); if any major functionality was changed or introduced
- [ ] [k8s LocalNet](https://github.com/pokt-network/pocket/blob/main/build/localnet/README.md); if any infrastructure or configuration changes were made

## Required Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added, or updated, [`godoc` format comments](https://go.dev/blog/godoc) on touched members (see: [tip.golang.org/doc/comment](https://tip.golang.org/doc/comment))
- [ ] I have tested my changes using the available tooling
- [ ] I have updated the corresponding CHANGELOG

### If Applicable Checklist

- [ ] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

File: docs/development/CODE_REVIEW_GUIDELINES.md [hunk 1]
```diff
@@ -57,6 +57,17 @@ Reviewers (and prospective reviewers) are encouraged to engage in reviews of cod
...
+tl;dr Use SOS, TECHDEBT(XXX), TECHDEBT, NIT if you think it'll help the author get context on next steps.
+
+When you leave reviews, if it is unclear from the reviewer to the author, please make sure to mention if:
+
+- NIT: Comment is a nitpick
+- TECHDEBT: Comment should have a TECHDEBT comment w/o a ticket
+- TECHDEBT(XXX): Comment should have a TECHDEBT comment but imporant enough that it requires a ticket to track the work in the near future
+- SOS: Show Stopper. You feel strongly it needs to be addressed now
+
+Only use the above if you think normal comments / TODOs are not self explanatory enough.
```
Comment (Olshansk) [inline docs/development/CODE_REVIEW_GUIDELINES.md:69]:
```suggestion
_tl;dr Use `SOS`, `TECHDEBT(XXX)`, `TECHDEBT`, `NIT` if you think it'll help the author get context on the next steps._

When you leave reviews, if it is unclear from the reviewer to the author, please make sure to mention if:

- **NIT**: Comment is a nitpick
- **TECHDEBT**: Comment should have a TECHDEBT comment w/o a ticket
- **TECHDEBT(XXX)**: Comment should have a TECHDEBT comment but important enough that it requires a ticket to track the work in the near future
- **SOS**: Show Stopper. You feel strongly it needs to be addressed now

Only use the above if you think normal comments / TODOs are not self-explanatory enough.
```

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: docs/development/CODE_REVIEW_GUIDELINES.md [hunk 2]
```diff
@@ -57,6 +57,17 @@ Reviewers (and prospective reviewers) are encouraged to engage in reviews of cod
 
 ### Reviewing
 
+tl;dr Use SOS, TECHDEBT(XXX), TECHDEBT, NIT if you think it'll help the author get context on next steps.
+
+When you leave reviews, if it is unclear from the reviewer to the author, please make sure to mention if:
```
Comment (bryanchriswhite) [inline docs/development/CODE_REVIEW_GUIDELINES.md:62]: I'm struggling a bit with this sentence. Am I interpreting it correctly to be roughly equivalent to something like the following?:

```suggestion
When leaving review comments, consider if any of the following characterizations applies and prefix the comment, respectively:
```

File: docs/development/CODE_REVIEW_GUIDELINES.md [hunk 3]
```diff
@@ -57,6 +57,17 @@ Reviewers (and prospective reviewers) are encouraged to engage in reviews of cod
 
 ### Reviewing
 
+tl;dr Use SOS, TECHDEBT(XXX), TECHDEBT, NIT if you think it'll help the author get context on next steps.
```
Comment (bryanchriswhite) [inline docs/development/CODE_REVIEW_GUIDELINES.md:60]: NIT: I fee like I always see it capitalized

```suggestion
TL;DR Use SOS, TECHDEBT(XXX), TECHDEBT, NIT if you think it'll help the author get context on next steps.
```

**Review by bryanchriswhite — CHANGES_REQUESTED** [review]
Thanks @dylanlott! :rocket: 

It's nice to see this thing start to accumulate some more content. :raised_hands:

I feel like it could be moved closer to the section which contains the other sentence which talks about review comments specifically: "During review, submit feedback using line comments..." (i.e. a but further down). I don't think the two sentences necessarily belong right next to each other but I feel like there is a bit of a readability improvement we could get by restructuring the similar bits to be closer together. Wdyt?
